apps/website/docs/infrastructure/migrate-opex.md

```diff
@@ -0,0 +1,113 @@
+when migrating to a new Azure Application Gateway.
+
+## Steps to Recreate Dashboards and Alerts
+
+Unlike any other Terraform resource, Opex dashboards and alerts are created
+using a custom Python application, and therefore Terraform source code is not
+available to the user. Therefore, the quickest way to delete programmatically
+existing dashboards and alerts is to create an empty Terraform configuration and
+apply it. This action will remove all existing dashboards and alerts, allowing
+you to recreate them with the new Application Gateway.
+
+1. Create a new directory for an empty Terraform configuration:
+
+```bash
+mkdir empty-terraform-config
+cd empty-terraform-config
+```
+
+2. Create a new file named `main.tf` with the following content:
+
+```hcl
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "~> 4.0"
+    }
+  }
+
+  backend "azurerm" {
+    resource_group_name  = "<your-resource-group-name>"
+    storage_account_name = "<your-storage-account-name>"
+    container_name       = "<your-container-name>"
+    key                  = "<your-terraform-state-file-name>"
+    use_azuread_auth     = true
+  }
+}
+
+provider "azurerm" {
+  features {
+  }
+  storage_use_azuread = true
+}
+```
+
+3. Replace the placeholders with your actual Azure resource names. They can be
+   found in the `backend.tfvars` file of the `opex` configuration.
+
+:::warning
+
+Be careful. Setting the wrong values could lead to the deletion of unwanted
+resources. **Double-check that your backend configuration matches the Opex state
+only.**
+
+:::
+
+4. Generate a Terraform plan to a file using the command:
+
+```bash
+terraform plan -no-color > tf.plan
+```
+
+5. Check the generated plan file to ensure that it only contains the deletion of
+   your Opex dashboard and alerts. **Do not proceed if other resources are
+   listed for deletion.**
+
+> _A correct plan should only show deletions for Opex dashboards and alerts.
+> Example:_
+>
+> ```hcl
+> # azurerm_monitor_scheduled_query_rules_alert.alarm_availability_0 will be destroyed
+> # (because azurerm_monitor_scheduled_query_rules_alert.alarm_availability_0 is not in configuration)
+> - resource "azurerm_monitor_scheduled_query_rules_alert" "alarm_availability_0" {
+> [...]
+> # azurerm_portal_dashboard.this will be destroyed
+> # (because azurerm_portal_dashboard.this is not in configuration)
+> - resource "azurerm_portal_dashboard" "this" {
+> ```
+
+6. If the plan looks correct, apply it using the command:
+
+```bash
+terraform apply tf.plan
+```
+
+7. After applying the empty configuration, all existing Opex dashboards and
+   alerts will be deleted. You can now update your Opex configuration with the
+   new Application Gateway id and run the GitHub Actions workflow to recreate
+   dashboards and alerts.
+
+:::info
+
+Due to internal limitations, the Opex resources will always be created in the
```

Comment:
suggestion: may you clarify "internal limitations" (of what?)?

Reply:
Values are hardcoded inside the application, but I don't think it's relevant for the reader